src/css/Parser.js

```diff
@@ -749,6 +755,92 @@ Parser.prototype = function(){
 
             },
 
+            _document: function(){
+                /*
+                 * document
+                 *   : DOCUMENT_SYM S*
+                 *     _document_function [ ',' S* _document_function ]* S*
+                 *     '{' S* ruleset* '}'
+                 *   ;
+                 */
+
+                var tokenStream = this._tokenStream,
+                    token,
+                    tt,
+                    functions = [],
+                    prefix = "";
+
+                tokenStream.mustMatch(Tokens.DOCUMENT_SYM);
+                token = tokenStream.token();
+                if (/^@\-([^\-]+)\-/.test(token.value)) {
+                    prefix = RegExp.$1;
+                }
+
+                this._readWhitespace();
+                functions.push(this._document_function());
+
+                while(tokenStream.match(Tokens.COMMA)) {
+                    this._readWhitespace();
+                    functions.push(this._document_function());
+                }
+
+                tokenStream.mustMatch(Tokens.LBRACE);
+                this._readWhitespace();
+
+                this.fire({
+                    type:      "startdocument",
+                    functions: functions,
+                    prefix:    prefix,
+                    line:      token.startLine,
+                    col:       token.startCol
+                });
+
+                while(true) {
+                    if (tokenStream.peek() == Tokens.PAGE_SYM){
+                        this._page();
+                    } else if (tokenStream.peek() == Tokens.FONT_FACE_SYM){
+                        this._font_face();
+                    } else if (tokenStream.peek() == Tokens.VIEWPORT_SYM){
+                        this._viewport();
+                    } else if (tokenStream.peek() == Tokens.MEDIA_SYM){
+                        this._media();
+                    } else if (!this._ruleset()){
```

Comment:
Fails on `@keyframes`, should also check for `Tokens.KEYFRAMES_SYM` and execute `this._keyframes()`, here's my take:

```js
                var tokenAction = {};
                tokenAction[Tokens.PAGE_SYM] = this._page;
                tokenAction[Tokens.FONT_FACE_SYM] = this._font_face;
                tokenAction[Tokens.VIEWPORT_SYM] = this._viewport;
                tokenAction[Tokens.MEDIA_SYM] = this._media;
                tokenAction[Tokens.KEYFRAMES_SYM] = this._keyframes;

                while(true) {
                    var action = tokenAction[tokenStream.peek()];
                    if (action) {
                    	action.call(this);
                    } else if (!this._ruleset()) {
                        break;
                    }
                }

                token = tokenStream.token();
                tokenStream.mustMatch(Tokens.RBRACE);
                this._readWhitespace();

                this.fire({
..............
```

Now I can submit this as a PR if you don't mind.